nemo/collections/nlp/data/language_modeling/megatron/lm_adapted_t5_dataset.py

```diff
@@ -0,0 +1,73 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+
+from nemo.collections.nlp.data.language_modeling.megatron.gpt_dataset import GPTDataset
+
+
+class T5LMAdaptedDataset(GPTDataset):
+    """
+    Dataset for unlearning span corruption (https://arxiv.org/abs/2104.08691) in T5 models.
+    Corresponds to the prefix-LM objective in the T5 paper (Table 3 in https://arxiv.org/abs/1910.10683).
+    """
+
+    def __init__(
+        self, cfg, trainer, tokenizer, name, data_prefix, documents, indexed_dataset, num_samples, seed, **kwargs
+    ):
+        self.seq_length_encoder = cfg.data.seq_length
+        self.seq_length_decoder = cfg.data.seq_length_dec
```

Comment:
I like the separation of `seq_length_dec`, nice!
Perhaps we can set it to `seq_length` if it is None, or negative (which will be the default)?

Reply:
I think it's better to set it to be the same as in the config. The reason being that the typical usecase is to fine-tune a pre-trained T5 on this LM objective. If a user ends up providing a seq_length_dec > pre-training decoder sequence length, the learnable position embeddings will fail. We can modify this once we add relative positional encodings maybe.

Reply:
I missed that was in the fine-tuning script. What is the default for the pre-training script? Is `seq_length_dec` has to be provided?

Reply:
Yes, it needs to be provided in the pre-training script. The default in the yaml config is 128.

Reply:
Ok perfect. That could be overridden for other models easily.